gms/impl/src/main/java/com/linkedin/metadata/resources/entity/EntityResource.java

```diff
@@ -121,32 +128,46 @@
 
   @Action(name = ACTION_INGEST)
   @Nonnull
-  public Task<Void> ingest(@ActionParam(PARAM_ENTITY) @Nonnull Entity entity) throws URISyntaxException {
+  public Task<Void> ingest(
+      @ActionParam(PARAM_ENTITY) @Nonnull Entity entity,
+      @ActionParam(SYSTEM_METADATA) @Optional @Nullable SystemMetadata systemMetadata
+  ) throws URISyntaxException {
+
+    if (systemMetadata == null) {
+      SystemMetadata generatedSystemMetadata = new SystemMetadata();
+      generatedSystemMetadata.setLastObserved(System.currentTimeMillis());
+      systemMetadata = generatedSystemMetadata;
+    }
+
     final Set<String> projectedAspects = new HashSet<>(Arrays.asList("browsePaths"));
     final RecordTemplate snapshotRecord = RecordUtils.getSelectedRecordTemplateFromUnion(entity.getValue());
     final Urn urn = com.linkedin.metadata.dao.utils.ModelUtils.getUrnFromSnapshot(snapshotRecord);
 
-    log.info("INGEST urn {}", urn.toString());
+    log.info("INGEST urn {} with system metadata {}", urn.toString(), systemMetadata.toString());
 
     final Entity browsePathEntity = _entityService.getEntity(urn, projectedAspects);
     BrowsePathUtils.addBrowsePathIfNotExists(entity.getValue(), browsePathEntity);
 
     // TODO Correctly audit ingestions.
     final AuditStamp auditStamp =
         new AuditStamp().setTime(_clock.millis()).setActor(Urn.createFromString(DEFAULT_ACTOR));
+    SystemMetadata finalSystemMetadata = systemMetadata;
     return RestliUtils.toTask(() -> {
-      _entityService.ingestEntity(entity, auditStamp);
+      _entityService.ingestEntity(entity, auditStamp, finalSystemMetadata);
       return null;
     });
   }
 
   @Action(name = ACTION_BATCH_INGEST)
   @Nonnull
-  public Task<Void> batchIngest(@ActionParam(PARAM_ENTITIES) @Nonnull Entity[] entities) throws URISyntaxException {
+  public Task<Void> batchIngest(
+      @ActionParam(PARAM_ENTITIES) @Nonnull Entity[] entities,
```

Comment:
Can we validate that the list of entities is the same size as the list of system metadata?

Reply:
will add 👍

Reply:
added

Reply:
what if systemMetadataList is null?

Reply:
@gabe-lyons

Reply:
Good call- I'll generate system metadata if not provided